google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/vcs/CloudRepositoryService.java

```diff
@@ -45,29 +46,38 @@
   private static int LIST_TIMEOUT_MS = 5000;
 
   public ListReposResponse list(CredentialedUser user, String cloudProject)
-      throws IOException, GeneralSecurityException {
-    Credential credential = user.getCredential();
-    HttpRequestInitializer initializer = httpRequest -> {
-      HttpHeaders headers = new HttpHeaders();
-      httpRequest.setConnectTimeout(LIST_TIMEOUT_MS);
-      httpRequest.setReadTimeout(LIST_TIMEOUT_MS);
-      httpRequest.setHeaders(headers);
-      credential.initialize(httpRequest);
-    };
-
-    HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-    String userAgent = ServiceManager
-        .getService(CloudToolsPluginInfoService.class).getUserAgent();
-
-    Source source = new Source.Builder(httpTransport, JacksonFactory.getDefaultInstance(),
-        initializer)
-        .setRootUrl(CLOUD_SOURCE_API_ROOT_URL)
-        .setServicePath("")
-        // this ends up prefixed to user agent
-        .setApplicationName(userAgent)
-        .build();
-
-    return new CustomUrlSourceRequest(source, cloudProject).execute();
+      throws CloudRepositoryServiceException {
+    try {
+      Credential credential = user.getCredential();
+      HttpRequestInitializer initializer = httpRequest -> {
+        HttpHeaders headers = new HttpHeaders();
+        httpRequest.setConnectTimeout(LIST_TIMEOUT_MS);
+        httpRequest.setReadTimeout(LIST_TIMEOUT_MS);
+        httpRequest.setHeaders(headers);
+        credential.initialize(httpRequest);
+      };
+
+      HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
```

Comment:
The API makes us go this low? I wonder if the new one takes care of this for us..